views/js/runner/plugins/navigation/next.js

```diff
@@ -96,41 +101,85 @@ define([
+                    endTestWarning:     context.options.endTestWarning,
+                    isLast:             context.isLast,
+                    isLinear:           context.isLinear,
+                    nextItemWarning:    nextItemWarning,
+                    nextPart:           mapHelper.getItemPart(map, nextItemPosition),
+                    remainingAttempts:  context.remainingAttempts,
+                    testPartId:         context.testPartId
+                });
 
-                e.preventDefault();
+                function enable() {
+                    testRunner.trigger('enablenav enabletools');
+                }
 
-                if(self.getState('enabled') !== false){
-                    self.disable();
-                    if(context.options.endTestWarning && context.isLast){
+                if(self.getState('enabled') !== false) {
+                    testRunner.trigger('disablenav disabletools');
+
+                    if (warningHelper.shouldWarnBeforeEnd()) {
                         testRunner.trigger(
                             'confirm.endTest',
                             messages.getExitMessage(
                                 __('You are about to submit the test. You will not be able to access this test once submitted. Click OK to continue and submit the test.'),
                                 'test', testRunner),
-                            _.partial(doNext, context), // if the test taker accept
+                            _.partial(triggerNextAction, context), // if the test taker accept
+                            enable  // if the test taker refuse
+                        );
+
+                    } else if (warningHelper.shouldWarnBeforeNext()) {
+                        testRunner.trigger(
+                            'confirm.next',
+                            __('You are about to go to the next item. Click OK to continue and go to the next item.'),
+                            _.partial(triggerNextAction, context), // if the test taker accept
                             enable  // if the test taker refuse
                         );
+
                     } else {
-                        doNext(context);
+                        triggerNextAction(context);
                     }
                 }
+            }
+
+            function triggerNextAction(context) {
+                if(context.isLast){
+                    self.trigger('end');
+                }
+                testRunner.next();
+            }
+
+            this.$element.on('click', function(e){
+                e.preventDefault();
+                testRunner.trigger('nav-next');
             });
 
+            if(testConfig.allowShortcuts && pluginShortcuts.toggle){
+                shortcut.add(namespaceHelper.namespaceAll(pluginShortcuts.toggle, this.getName(), true), function(e) {
+                    if (self.getState('enabled') === true) {
+                        e.preventDefault();
```

Comment:
This instruction is not needed here, the shortcut helper already does this by default.
At least you should use the `prevent` option to be sure the event is prevented.